pkg/storage/bucket/s3/bucket_client.go

```diff
@@ -12,6 +12,11 @@ import (
 	"github.com/thanos-io/objstore/providers/s3"
 )
 
+const (
+	// Applied to PUT operations to denote the desired storage class for S3 Objects
+	awsStorageClassHeader = "X-Amz-storage-Class"
```

Comment:
[nit] Could you please make the casing consistent here?
```suggestion
	awsStorageClassHeader = "X-Amz-Storage-Class"
```